File: complexnumber.ts
const complexNumberMultiply = (num1: string, num2: string): string => {

  const complexToArray = (num: string): number[] => {
    const [ real, imaginary ] = num.split("+");
    return new Array<number>(+real, +imaginary.split("i")[0]);
  }

  const complexToString = (real: number, imaginary: number): string => {
    return `${real}+${imaginary}i`
  }

  const [a, b] = complexToArray(num1);
  const [c, d] = complexToArray(num2);
  const real = a*c - b*d;
  const imaginary = a*d + b*c;

  return complexToString(real, imaginary);
};

File: complexnumber.py
# 537. Complex Number Multiplication
# https://leetcode.com/problems/complex-number-multiplication/

def complexNumberMultiply(num1: str, num2: str) -> str:

  def to_tuple(num: str) -> tuple[int]:
    real, imaginary = num.split("+")
    return int(real), int(imaginary.split("i")[0])

  def to_string(real: int, imaginary: int) -> str:
    return f"{real}+{imaginary}i"

  a, b = to_tuple(num1)
  c, d = to_tuple(num2)
  real, imaginary = (a*c - b*d), (a*d + b*c)
  return to_string(real, imaginary)
